### PR TITLE
feat(Listbox): keyboard nav on hover

### PR DESCRIPTION
**Pull request by DanielS-Qlik**
## Motivation

This PR will enable keyboard navigation when hovering the Listbox (without entering selection state).
Kapture 2023-03-02 at 16 44 22

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [x] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [x] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 1]
```diff
@@ -123,8 +125,12 @@ export default function ListBox({
     local.current.validPages = false;
     if (loaderRef.current) {
       loaderRef.current.resetloadMoreItemsCache(true);
+      const isScrollingToEnd = keyScroll.state.scrollPosition === 'overflowEnd';
       // Skip scrollToItem if we are in selections.
-      if (layout && layout.qSelectionInfo.qInSelections) {
+      if ((layout && layout.qSelectionInfo.qInSelections) || isScrollingToEnd) {
+        if (isScrollingToEnd) {
+          keyScroll.reset();
+        }
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/ListBox.jsx:133]:
```suggestion
      // Skip scrollToItem if we are in selections or if scrolling to the end.
      if ((layout && layout.qSelectionInfo.qInSelections) || isScrollingToEnd) {
        keyScroll.reset();
```

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 2]
```diff
@@ -193,6 +199,21 @@ export default function ListBox({
...
+    } else if (s.down) {
+      scrollToIndex(currentScrollIndex.state.stop + s.down);
+    } else if (s.scrollPosition) {
+      setScrollPosition(s.scrollPosition);
+    }
+    if (s.scrollPosition === 'overflowEnd') {
+      return; // Do keyScroll.reset() in fetchData() to avoid scrolling to top.
+    }
+    keyScroll.reset();
+  }, [JSON.stringify(keyScroll.state)]);
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/ListBox.jsx:215]: Don't know if it's any better… you decide:
```suggestion
  }, [Object.values(keyScroll.state)]);
```

File: apis/nucleus/src/components/listbox/__tests__/list-box-portal.test.jsx [hunk 3]
```diff
@@ -131,6 +131,7 @@ describe('ListBoxPortal', () => {
             ...options,
             selections: options.selectionsApi,
             model: options.sessionModel,
+            app: options.app,
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/__tests__/list-box-portal.test.jsx:134]:
```suggestion
```

File: apis/nucleus/src/components/listbox/ListBox.jsx (hunk 2 above)
Comment (DanielS-Qlik) [reply apis/nucleus/src/components/listbox/ListBox.jsx]: Yes that's probably best, it just hurts my dx eyes a bit.

File: apis/nucleus/src/components/listbox/__tests__/list-box-portal.test.jsx (hunk 3 above)
Comment (DanielS-Qlik) [reply apis/nucleus/src/components/listbox/__tests__/list-box-portal.test.jsx]: I think I made the test a bit easier to read.

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 4]
```diff
@@ -193,6 +199,21 @@ export default function ListBox({
...
+    if (s.up) {
+      scrollToIndex(currentScrollIndex.state.start - s.up);
+    } else if (s.down) {
+      scrollToIndex(currentScrollIndex.state.stop + s.down);
+    } else if (s.scrollPosition) {
+      setScrollPosition(s.scrollPosition);
+    }
+    if (s.scrollPosition === 'overflowEnd') {
+      return; // Do keyScroll.reset() in fetchData() to avoid scrolling to top.
+    }
```
Comment (johanlahti) [inline apis/nucleus/src/components/listbox/ListBox.jsx:213]: Could use a `switch` instead here.

**Review by johanlahti — APPROVED** [review]

**Pull request merged**